src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/BuilderDesignPattern.java

```diff
@@ -0,0 +1,30 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+public class BuilderDesignPattern extends DesignPattern {
```

Comment:
If many design patterns extend an abstract base class we get a proliferation (20-30 or more) of design pattern concrete sub-classes, when the code base gets larger, adding more functionality could result in code duplication, breaking OCP and LSP. On the other hand if the designed module is not a subject to major changes this approach could be fine for example if the concrete design pattern classes were declared final.